labs-ios-starter/Application/Persistence/CoreDataManager.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2020 Shawn James. All rights reserved.
+// CoreDataManager.swift
+
+import CoreData
+
+/// Used to interact with persisted managed objects and the application's xcdatamodeld
+class CoreDataManager {
+    static let shared = CoreDataManager() // singleton
+    
+    // persistent container
+    lazy var persistentContainer: NSPersistentContainer = {
+        let persistentContainer = NSPersistentContainer(name: "labs-ios-starter")
+        persistentContainer.loadPersistentStores { (_, error) in
+            if let error = error {
+                fatalError("Failed to load persistent stores: \(error)")
+            }
+        }
+        return persistentContainer
+    }()
+    
+    // view context
+    var mainContext: NSManagedObjectContext {
+        return persistentContainer.viewContext
+    }
+    
+    // MARK: - Methods
+    
+    /// Saves to the main context with error handling.
+    func save() {
```

Comment:
Good thought adding the save function here, most people don't think of it.

I typically leave this function flexible, but have it default to the mainContext `func save(context: NSManagedObjectContext = mainContext)`. That way you can use it with different contexts without having to rewrite the function.